source/connectButtonToService.js

```diff
@@ -25,23 +38,15 @@ UpdateBroadcaster.prototype.trigger = function (value) {
 
 /**
  * Find or create an appropriate instance of UpdateBroadcaster
- *
- * @param {String} serviceName
- * @param {Object} options
  * @param {Function} buttonSetter
+ * @param {Object} options
  */
-
-export default (serviceName, buttonSetter, options) => {
-    if (!broadcastersForServices[serviceName]) {
-        broadcastersForServices[serviceName] = {};
-    }
-    const broadcastersForUrls = broadcastersForServices[serviceName];
-    let broadcaster = broadcastersForUrls[options.url];
-
-    if (!broadcaster || options.forceUpdate) {
-        broadcaster = new UpdateBroadcaster(options.counterUrl, options.url);
-        services[serviceName].fetch(broadcaster);
-        broadcastersForUrls[options.url] = broadcaster;
+export default (buttonSetter, options) => {
```

Comment:
Broadcasters(an entity with exact URL, counter value, and set of callbacks for the buttons which need the same value) now stored with the corresponding service.